hail/src/main/scala/is/hail/cxx/EmitTriplet.scala

```diff
@@ -98,77 +98,62 @@ abstract class ArrayEmitter(val setup: Code, val m: Code, val setupLen: Code, va
 }
 
 object NDArrayLoopEmitter {
-  def linearizeIndices(fb: FunctionBuilder, idxs: Seq[Variable], strides: Code, shape: Code): Code = {
-    val result = fb.variable("result", "int", "0")
-    val nDims = idxs.length
-    val buildIndex = idxs.zipWithIndex.map { case (idx, dim) =>
-        s"""
-           | if ($idx < 0 || $idx >= $shape[$dim]) {
-           |   throw new FatalError("Invalid index");
-           | }
-           |
-           | $result += $idx * $strides[$dim];
-         """.stripMargin
-    }.mkString("\n")
+  def linearizeIndices(idxs: Seq[Variable], strides: Code): Code = {
+    idxs.zipWithIndex.foldRight("0") { case ((idxVar, dim), linearIndex) =>
```

Comment:
Let's wrap this in `()` so that it's always guaranteed to be treated as a single expression?